### PR TITLE
Static analysis warnings

### PR DESCRIPTION
**Pull request by bluca**
@TomMD it seems the tool doesn't see when variables are used for array sizes, these are false positives:

1. DEAD_STORE
    - At [src/tcp_address.cpp(121,5)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/tcp_address.cpp#L121)
    - The value written to &max_port_str_length (type unsigned long const ) is never used.
1. DEAD_STORE
    - At [src/tcp_address.cpp(288,5)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/tcp_address.cpp#L288)
    - The value written to &max_mask_len (type unsigned long const ) is never used.

These are also false positives, the struct is initialized by the systemcall:

1. UNINITIALIZED_VALUE
    - At [src/stream_engine.cpp(79,13)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/stream_engine.cpp#L79)
    - The value read from cred.gid was never initialized.
1. UNINITIALIZED_VALUE
    - At [src/stream_engine.cpp(79,13)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/stream_engine.cpp#L79)
    - The value read from cred.pid was never initialized.
1. UNINITIALIZED_VALUE
    - At [src/stream_engine.cpp(79,13)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/stream_engine.cpp#L79)
    - The value read from cred.uid was never initialized.

These are also false positives:

1. UNINITIALIZED_VALUE
    - At [src/tweetnacl.c(138,15)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/tweetnacl.c#L138)
    - The value read from w[_] was never initialized.
1. UNINITIALIZED_VALUE
    - At [src/tweetnacl.c(149,7)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/tweetnacl.c#L149)
    - The value read from x[_] was never initialized.
1. UNINITIALIZED_VALUE
    - At [src/tweetnacl.c(575,18)](https://github.com/zeromq/libzmq/blob/90ff30c086bc299d619346a04d9eb5a02e96636e/src/tweetnacl.c#L575)
    - The value read from a[_] was never initialized.

### CODE REVIEW

**Comment by bluca** [comment]
Can a file be ignored? We don't want to modify external sources (unity source code)

**Comment by TomMD** [comment]
> @TomMD it seems the tool doesn't see when variables are used for array sizes, these are false positives:

Yes I see.  The information is lost by clang, making the dead store that isn't there in source appear in the bitcode.  I am considering solutions.

> These are also false positives, the struct is initialized by the systemcall:
> ...
> These are also false positives:
> ...

While there will always be some false positives they needn't be displayed in every pull request.  If the bug is not related to the pull request then it appears in the summary.  We are considering moving the summary to console.muse.dev via a hyperlinking the text `Muse Analysis Complete`.  Other solutions are possible and I'm trying to understand what is most preferable.

> Can a file be ignored? We don't want to modify external sources (unity source code)

I can add that capability.  The tool options are placed in repo under `.muse/config` for example:

```
ignore = [ "DEAD_STORE", "UNINITIALIZED_VALUE"]
```

It should become a `config.toml` file, I believe, and have another option for ignored files (vs bugs) making the configuration for libzmq something like:

```
ignoreBugs = [ "DEAD_STORE", "UNINITIALIZED_VALUE" ]
ignoreFiles = [ "*tweetnacl*", "*/unity.c" ] # Design Q for team: regex vs bash globbing vs full paths
```

We can have this ready for you guys soon.  I'll close with the team on a design.

**Comment by bluca** [comment]
Thanks - something else that should really be done is moving the output in the "checks" section with the other CI tools, rather than having comments, as they are quite annoying (they spam notifications)

**Comment by sigiesec** [comment]
At least they are annoying since they refer to issues that are already present on master. If they were only new issues, the commenting behaviour might be acceptable.

**Comment by bluca** [comment]
@sigiesec can this be merged in the meanwhile? These fixes are for valid warnings

File: src/udp_engine.cpp [hunk 1]
```diff
@@ -396,8 +396,8 @@ void zmq::udp_engine_t::out_event ()
     if (rc == 0) {
         msg_t body_msg;
         rc = _session->pull_msg (&body_msg);
-        //  TODO rc is not checked here. We seem to assume rc == 0. An
-        //  assertion should be added.
+        //  If there's a group, there should also be a body
+        errno_assert (rc == 0);
```
Comment (bluca) [inline src/udp_engine.cpp]: @somdoron FYI I think this is the right thing here, but correct me if I'm wrong

**Pull request merged**